common/src/main/java/org/conscrypt/Conscrypt.java

```diff
@@ -75,12 +75,47 @@ public static Provider newProvider() {
 
     /**
      * Constructs a new {@link Provider} with the given name.
+     *
+     * @deprecated Use {@link #newProviderBuilder()} instead.
      */
+    @Deprecated
     public static Provider newProvider(String providerName) {
         checkAvailability();
         return new OpenSSLProvider(providerName);
     }
 
+    public static class ProviderBuilder {
+        private String name;
+        private boolean provideTrustManager = false;
+
+        private ProviderBuilder() {}
+
+        /**
+         * Sets the name of the Provider to be built.
+         */
+        public ProviderBuilder setName(String name) {
+            this.name = name;
+            return this;
+        }
+
+        /**
+         * Causes the returned provider to provide an implementation of
+         * {@link javax.net.ssl.TrustManagerFactory}.
+         */
+        public ProviderBuilder provideTrustManager() {
+            this.provideTrustManager = true;
+            return this;
+        }
+
+        public Provider build() {
+            return new OpenSSLProvider(name, provideTrustManager);
+        }
+    }
+
+    public static ProviderBuilder newProviderBuilder() {
+        return new ProviderBuilder();
```

Comment:
Name is required .... should throw if it's not set.

Reply:
I made it default to the default name, so people don't have to set it if they don't need to customize it.